src/main/java/org/junitpioneer/jupiter/json/Param.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.json;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation indicating the name of the parameter that should be extracted into the method.
+ * This is needed when the test code is not compiled with Java 8 {@code -parameters} flag.
+ *
+ * @since TBD
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface Param {
```

Comment:
This name should probably be a bit longer - see [related discussion](https://github.com/junit-pioneer/junit-pioneer/issues/415#issuecomment-861770281) on Cartesian tests.

Reply:
I will change it to `Parameter`. If you have better idea let me know